src/subdomains/staking/application/services/staking-deposit.service.ts

```diff
@@ -216,17 +220,17 @@ export class StakingDepositService {
     return staking.verifyUserAddresses(addresses);
   }
 
-  private createOrUpdateDeposit(staking: Staking, payIn: PayIn): void {
-    const deposit = staking.getDepositByPayInTxId(payIn.txId) ?? this.createNewDeposit(staking, payIn);
+  private async createOrUpdateDeposit(staking: Staking, payIn: PayIn): Promise<void> {
+    const deposit =
+      (await this.depositRepository.getByPayInTxId(staking.id, payIn.txId)) ??
+      (await this.createNewDeposit(staking, payIn));
 
     deposit.updatePreCreatedDeposit(payIn.txId, payIn.amount);
   }
 
-  private createNewDeposit(staking: Staking, payIn: PayIn): Deposit {
+  private async createNewDeposit(staking: Staking, payIn: PayIn): Promise<Deposit> {
     const newDeposit = this.factory.createDeposit(staking, { amount: payIn.amount, txId: payIn.txId });
 
-    staking.addDeposit(newDeposit);
-
-    return newDeposit;
+    return this.depositRepository.save(newDeposit);
   }
```

Comment:
I think the `createNewDeposit` should not save the deposit. It needs to be saved only after line 228 (for create and update)?

Reply:
you are absolutely right, my bad...